packages/edit-site/src/components/global-styles/variations-typography.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+import {
+	__experimentalGrid as Grid,
+	__experimentalVStack as VStack,
+	__unstableMotion as motion,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { mergeBaseAndUserConfigs } from './global-styles-provider';
+import { unlock } from '../../lock-unlock';
+import { getFamilyPreviewStyle } from './font-library-modal/utils/preview-styles';
+import { useCurrentMergeThemeStyleVariationsWithUserConfig } from '../../hooks/use-theme-style-variations/use-theme-style-variations-by-property';
+import Subtitle from './subtitle';
+import Variation from './variation';
+
+const { GlobalStylesContext } = unlock( blockEditorPrivateApis );
+
+function getFontFamilyFromSetting( fontFamilies, setting ) {
+	if ( ! setting ) {
+		return null;
+	}
+
+	const fontFamilyVariable = setting.replace( 'var(', '' ).replace( ')', '' );
+	const fontFamilySlug = fontFamilyVariable?.split( '--' ).slice( -1 )[ 0 ];
+
+	return fontFamilies.find(
+		( fontFamily ) => fontFamily.slug === fontFamilySlug
+	);
+}
+
+const getFontFamilies = ( themeJson ) => {
+	const fontFamilies = themeJson?.settings?.typography?.fontFamilies?.theme; // TODO this could not be under theme.
+	const bodyFontFamilySetting = themeJson?.styles?.typography?.fontFamily;
+	const bodyFontFamily = getFontFamilyFromSetting(
+		fontFamilies,
+		bodyFontFamilySetting
+	);
+
+	const headingFontFamilySetting =
+		themeJson?.styles?.elements?.heading?.typography?.fontFamily;
+
+	let headingFontFamily;
+	if ( ! headingFontFamilySetting ) {
+		headingFontFamily = bodyFontFamily;
+	} else {
+		headingFontFamily = getFontFamilyFromSetting(
+			fontFamilies,
+			themeJson?.styles?.elements?.heading?.typography?.fontFamily
+		);
+	}
+
+	return [ bodyFontFamily, headingFontFamily ];
+};
+const getFontFamilyNames = ( themeJson ) => {
+	const [ bodyFontFamily, headingFontFamily ] = getFontFamilies( themeJson );
+	return [ bodyFontFamily?.name, headingFontFamily?.name ];
+};
+
+const TypePreview = ( { variation } ) => {
+	const { base } = useContext( GlobalStylesContext );
+	const [ bodyFontFamilies, headingFontFamilies ] = getFontFamilies(
+		mergeBaseAndUserConfigs( base, variation )
+	);
+	const bodyPreviewStyle = bodyFontFamilies
+		? getFamilyPreviewStyle( bodyFontFamilies )
+		: {};
+	const headingPreviewStyle = headingFontFamilies
+		? getFamilyPreviewStyle( headingFontFamilies )
+		: {};
+	return (
+		<motion.div
```

Comment:
Same as color. I wear glasses, but I can't see any animation? Could we ditch motion here?


https://github.com/WordPress/gutenberg/assets/6458278/a7fb5a5f-14d6-4f5a-9557-8bc8a6d46abe

Reply:
Sorry, I didn't see the transition as the panel opens.

https://github.com/WordPress/gutenberg/assets/6458278/2b3ffaeb-814a-4512-aaf8-1f7fa90db3a5